### ISSUE TITLE
date/time not preserved with "cp -p" and "tar -xp"

### ISSUE THREAD

**Issue by amiga23** [juicedata/juicefs#3419]
**What happened**:
when copying files with "cp -p" or extracting tar archives with -p flag, the date/time are not preserved.

**What you expected to happen**:
date and time should be preserved when using -p flag

**How to reproduce it (as minimally and precisely as possible)**:
Use simple example from Getting Started Quickstart and copy a file with "cp -p" or extract an archive with "tar -xpf" in the mounted juicefs volume.
I've tested with cloud provider and local filesystem. As metadata I tested with sqlite and with local redis.

**Anything else we need to know?**
nothing special

**Environment**:
- JuiceFS version (use `juicefs --version`) or Hadoop Java SDK version: juicefs version 1.0.3+2022-12-27.e4bf15aca99
- Cloud provider or hardware configuration running JuiceFS: happens with local filesystem and with cloud provider
- OS (e.g `cat /etc/os-release`): Manjaro rolling
- Kernel (e.g. `uname -a`): Linux local 6.1.19-1-MANJARO #1 SMP PREEMPT_DYNAMIC Mon Mar 13 12:59:35 UTC 2023 x86_64 GNU/Linux
- Object storage (cloud provider and region, or self maintained): local filesystem
- Metadata engine info (version, cloud provider managed or self maintained): redis or sqlite
- Network connectivity (JuiceFS to metadata engine, JuiceFS to object storage): local
- Others:

**Comment by amiga23** [comment]
changing the date/time with "touch -m -t 01020404 myjfs/testfile" works

**Comment by SandyXSD** [comment]
Probably a duplication of #3204

**Comment by amiga23** [comment]
Yes, duplicate of https://github.com/juicedata/juicefs/issues/3204, sorry did somehow not see it.
